packages/app-builder/src/components/Cases/CaseEvents.tsx

```diff
@@ -298,10 +301,26 @@ function Author({
   );
 }
 
+function ByWorkflow({
+  type,
+}: {
+  type: 'added_by_workflow' | 'edited_by_workflow';
+}) {
+  const { t } = useTranslation(casesI18n);
+  return (
+    <div className="text-grey-100 text-s font-semibold">
+      <Trans t={t} i18nKey={`cases:case_detail.history.event_detail.${type}`} />
+    </div>
+  );
+}
+
 export function getEventDetail(event: CaseEvent) {
   const { event_type } = event;
   switch (event_type) {
     case 'case_created': {
+      if (!event.user_id) {
+        return <ByWorkflow type="added_by_workflow" />;
+      }
```

Comment:
With the `optionnal` thing, removing this line should raise a TS error here

Reply:
now does 👍